frontend/src/routes/login.lazy.tsx

```diff
@@ -52,52 +57,74 @@ function Login() {
           </p>
         </div>
 
-        <form
+        <Form.Root
           className="p-7 md:px-32 md:w-[50%]  flex flex-col w-full"
-          action={(formData) =>
-            mutate({
-              username: formData.get("username")!.toString(),
-              password: formData.get("password")!.toString()
-            })
-          }
+          asChild
         >
-          <h1 className="md:text-2xl text-3xl md:text-left text-center font-bold my-3">
-            Log in
-          </h1>
-          <div className="card flex flex-col gap-3">
-            {data?.errors?.root && (
-              <div style={{ color: "red" }}>{data.errors.root}</div>
-            )}
-            <div>
-              <Input
-                placeholder="username"
-                aria-label="username"
-                name="username"
-                type="text"
-              />
-              {!!data?.errors?.username && (
-                <p style={{ color: "red" }}>{data.errors.username}</p>
-              )}
-            </div>
-            <div>
-              <Input
-                aria-label="password"
-                placeholder="password"
-                type="password"
-                name="password"
-              />
-              {!!data?.errors?.password && (
-                <p style={{ color: "red" }}>{data.errors.password}</p>
+          <form
+            action={(formData) =>
+              mutate({
+                username: formData.get("username")!.toString(),
+                password: formData.get("password")!.toString()
+              })
+            }
+          >
+            <h1 className="md:text-2xl text-3xl md:text-left text-center font-bold my-3">
+              Log in
+            </h1>
+            <div className="card flex flex-col gap-3">
+              {data?.errors && (
+                <Alert variant="destructive">
+                  <AlertCircle className="h-4 w-4" />
+                  <AlertTitle>Error</AlertTitle>
+                  <AlertDescription>{data?.errors?.root}</AlertDescription>
```

Comment:
```suggestion
                  <AlertDescription>{data.errors.root}</AlertDescription>
```

No need to use exclamation points when you already checked for it, it is already defined and typescript won't yell at you